sdk/cosmosdb/cosmos/src/queryExecutionContext/orderByDocumentProducerComparator.ts

```diff
@@ -109,45 +109,43 @@ export class OrderByDocumentProducerComparator {
     return compFunc(item1, item2);
   }
 
-  public compareOrderByItem(orderByItem1: any, orderByItem2: any) {
+  private compareOrderByItem(orderByItem1: any, orderByItem2: any) {
     const type1 = this.getType(orderByItem1);
     const type2 = this.getType(orderByItem2);
     return this.compareValue(orderByItem1["item"], type1, orderByItem2["item"], type2);
   }
 
-  public validateOrderByItems(res1: string[], res2: string[]) {
-    this._throwIf(res1.length !== res2.length, `Expected ${res1.length}, but got ${res2.length}.`);
-    this._throwIf(
-      res1.length !== this.sortOrder.length,
-      "orderByItems cannot have a different size than sort orders."
-    );
+  private validateOrderByItems(res1: string[], res2: string[]) {
+    if (res1.length !== res2.length) {
+      throw new Error(`Expected ${res1.length}, but got ${res2.length}.`);
+    }
+    if (res1.length !== this.sortOrder.length) {
+      throw new Error("orderByItems cannot have a different size than sort orders.");
+    }
 
     for (let i = 0; i < this.sortOrder.length; i++) {
       const type1 = this.getType(res1[i]);
       const type2 = this.getType(res2[i]);
-      this._throwIf(type1 !== type2, `Expected ${type1}, but got ${type2}.`);
+      if (type1 !== type2) {
+        throw new Error(`Expected ${type1}, but got ${type2}.Cannot execute cross partition order - by queries on mix types.Consider filtering your query using IS_STRING / IS_NUMBER to get around this exception.`);
```

Comment:
Spaces after periods would be appreciated I think.

Reply:
Also probably good to make it `order-by` not `order - by`.

Reply:
Committed your suggestion